### ISSUE TITLE
Replace MessageBox with Toast notification for windows 8+?

### ISSUE THREAD

**Issue by kataras** [gen2brain/beeep#1]
Hello @gen2brain ,

First of all, I'm glad that I have the chance to welcome this library to the Go Community by opening the first issue here! You did a very good job, it's quite simple library but structured very well and you can extend it even more.

I was thinking instead of using the message box, if windows 8 or windows 10 then use the toast notification (right-side notification box, just like more window apps do it) because message box is good for showing errors and things like that but I think showing "information" as notification is the best option to go.

**Comment by gen2brain** [comment]
Hello, 

Thanks, the plan is to change that message box with some other component. I know nothing about Windows APIs so still not sure what is possible. It would be great to not use `cgo` for that, so far everything is in native Go. Something with syscall and user32, kernel32 etc., like https://github.com/lxn/win or https://github.com/lxn/walk would be perfect.  We doesn't have to use those libraries, just maybe pick some parts, it seems there are some parts that import C.

As I see, for standard "baloon" message it needs icon in system tray (notification area), and also a window (I guess can be hidden from start, or it is enough to not Show it), but doesn't make much sense to create all that every time notifications is sent, and if it is possible to use it like that.

**Comment by kataras** [comment]
Hey @gen2brain I tested the new box, it looks far modern than the previous and that's a progress. I can give you a good idea without need to do anything hard-core by hand: Find an executable written in any language that can run without any runtime framework, i.e C, which does toast notifications, we don't need cgo I agree, but there is another way: embedd that (small) executable file, the application) to your library with `go-bindata` or any other library that you're used to work and then if windows call that instead, this will be inside our applications as well no need to "download" anything, because it's "go code". I don't remember even the kernel, hope I could just PR the feature... but I am not working with any windows internals for a long time :/

**Comment by gen2brain** [comment]
Great! Thanks for suggestions,  I will check how it is implemented there. I would like to see first what are the options, including some binary will be last resort. For Web/GopherJS there is a hard coded beep sound, but only because there are no other options.

I learned something from every project I started btw. the whole reason why I start building something in the first place. Recently I feel like I upgraded to new level :) but there are always more levels ahead.
